chore(ia): Tidy MDN meta dir (#37784)
mdn/content: files/en-us/mdn/community/open_source_etiquette/index.md

{{MDNSidebar}}

If you've not worked on an open source project (OSP) before, it is a good idea to read this article before starting to contribute to MDN Web Docs and other open source projects.
There are a few behaviors to adopt that will help you and the other project contributors feel valued and safe, and stay productive.
This article won't teach you everything about contributing to open source; the aim is to cover foundational topics for taking part in open source communities.

## Think about why you are contributing to an OSP

Before you start contributing to an open source project, ask yourself why you want to do that.
It's fine if the answer to this question is "I want something to do with my time", but even better reasons might include:

- I want to improve my skills.
- I use this tool all the time and found a bug in it or I want to help make it better.
- I want to help other people use the tool more successfully.
- I want to help other people contribute to the project more successfully.
- I want to publicly demonstrate my own skills for my college or university course or to improve my chances of getting a job.

Some of these reasons are self-serving, and that's fine!

> Having a clear set of reasons for contributing will make it easier to decide what to work on first.
> Your presence on the project should be productive, and it shouldn't stop others from being so, too.

Reviewer: This seems better to me, but I don't want to explain why

```suggestion
Having a clear set of reasons for contributing will make you more productive and make it easier to work with the community.
```

This is our number one bit of advice for anyone starting open source contributions.
Be kind to the other contributors on the project, and it will be a happier and more productive place.

- Thank people if they help you.
- Congratulate people where appropriate, like if they land a pull request, or fix a difficult bug.
- Always respond respectfully, even if you feel like the answer to a question was obvious or that someone is repeating themselves.
- Try to help people to do better in a supportive way.
  For example, saying "this is wrong" or "here is the answer" is not as helpful as saying "This is OK, but I feel that it would be better if we did it more like this, here's a blog post for more ideas" or "you can find the answer here; also check out this link for more common answers".

Contributors are here because they want to make a positive impact on the project.
Beyond that, don't make assumptions, such as:

- Knowledge of the project and the technologies used to build it
- Gender, sexuality, age, languages spoken, location, political views, religion, or other personal attributes
- Experience with open source projects
- Confidence level
- Expectations
- Sense of humor

> You should keep what you write on topic and avoid controversial subjects like religion or politics.

Reviewer: ```suggestion
You should keep what you write on topic and avoid controversial subjects, such as religion or politics.
```

> Refrain from swearing or offensive language, even if it is not directed at anyone in particular.
> It is not needed for participation, and some people are sensitive to it.

Reviewer: ```suggestion
Refrain from swearing or potentially offensive language.
It rarely improves communication, and can make it harder for others to participate.
```

Be supportive and respectful even if you disagree with someone, or don't like a decision they've made.
Be aware that there are rules in place in any good OSP to protect its contributors against being made to feel uncomfortable while contributing.

> This usually can be found in a `CODE_OF_CONDUCT.md` file on GitHub.

Reviewer: Perhaps "See XXX for MDN's code of conduct"

You could also contribute by opening [pull requests](/en-US/docs/MDN/Community/Pull_requests) to fix problems that you come across while reading MDN articles.

A lot of the work on MDN revolves around writing documentation and code examples, but there are other ways to contribute too.
This could be helping to triage issues that come in, fixing typos, fixing grammar to make pages more understandable, or mentoring people that are trying to make fixes.
Every fix is useful, no matter how small, and we won't turn any away.
Nevertheless, try to make sure your fixes are productive. We'd like to advise against these kinds of contributions:

- Updating code style, language in prose, or testing framework because you like it better.
- Changing pages from US English to British English.
- Adding or removing punctuation when the original is correct.

In many cases, things are like they are on OSPs for a reason.
You should read style guides if they have one, and if in doubt about whether something is productive, always ask first!

## Read the manual

> Good OSPs will always make contributor documentation readily available. On GitHub projects, it is usually in the repo's CONTRIBUTING.md file, or sometimes in the project's README.md file. Being a documentation project, MDN content has a [README](https://github.com/mdn/content/blob/main/README.md) and a decent set of contributor docs on the site itself (see [Contributing to MDN Web Docs](/en-US/docs/MDN/Community)).

Reviewer: FYI "Contributing to MDN Web Docs" is what you now named "Community Resources".
